common-api/src/test/java/org/pmiops/workbench/db/model/CommonStorageEnumsTest.java

```diff
@@ -4,103 +4,86 @@
       assertThat(v).isEqualTo(CommonStorageEnums.domainIdToDomain(storageValue));
     }
   }
 
-  // copied from api/StorageEnumsTest because the above tests are not comprehensive
-  @Test
-  public void noMissingMapEntries() throws Exception {
-    for (Field f : CommonStorageEnums.class.getDeclaredFields()) {
-      if (f.getType() != BiMap.class) {
-        continue;
-      }
-
-      Class enumClass =
-          (Class) ((ParameterizedType) f.getAnnotatedType().getType()).getActualTypeArguments()[0];
-
-      Method enumToShort = null;
-      Method shortToEnum = null;
-      for (Method m : CommonStorageEnums.class.getDeclaredMethods()) {
-        if (m.getParameterTypes()[0].equals(enumClass)) {
-          enumToShort = m;
-        }
-
-        if (m.getReturnType().equals(enumClass)) {
-          shortToEnum = m;
-        }
-      }
-
-      // stringly typed map - test with testDomainIdBijectiveStorageMapping instead
-      if (enumToShort.getName().equals("domainIdToDomain")) {
-        continue;
-      }
+  // this is a copy of StorageEnumsTest.test_noMissingMapEntries()
 
+  @Test
+  public void test_noMissingMapEntries() throws Exception {
+    final Set<Class> enumClasses = getEnumerationClasses();
+    final Collection<Method> methods = Arrays.asList(CommonStorageEnums.class.getDeclaredMethods());
+
+    // e.g. public static Short domainToStorage(Domain domain)
+    final Map<Class, Method> toStorageMethods =
+        methods.stream()
+            .filter(m -> enumClasses.contains(firstMethodParameter(m)))
+            // domainToDomainId is stringly typed - test with test_domainId
+            .filter(m -> !m.getName().equals("domainToDomainId"))
+            .collect(Collectors.toMap(this::firstMethodParameter, m -> m));
+
+    // e.g. public static Domain domainFromStorage(Short domain)
+    final Map<Class, Method> fromStorageMethods =
+        methods.stream()
+            .filter(m -> enumClasses.contains(m.getReturnType()))
+            // domainToDomainId is stringly typed - test with test_domainId
+            .filter(m -> !m.getName().equals("domainIdToDomain"))
+            .collect(Collectors.toMap(Method::getReturnType, m -> m));
+
+    for (Class enumClass : enumClasses) {
+      final Method toStorage = toStorageMethods.get(enumClass);
+      final Method fromStorage = fromStorageMethods.get(enumClass);
       for (Object e : enumClass.getEnumConstants()) {
-        Short shortVal = (Short) enumToShort.invoke(null, e);
+        Short shortVal = (Short) toStorage.invoke(null, e);
 
         assertThat(shortVal).named(enumClass.getName() + ":" + e.toString()).isNotNull();
-        assertThat(shortToEnum.invoke(null, shortVal)).isEqualTo(e);
+        assertThat(fromStorage.invoke(null, shortVal)).isEqualTo(e);
       }
     }
   }
+
+  /**
+   * Retrieve all Enum classes in CommonStorageEnums
+   *
+   * <p>our convention for BiMaps in CommonStorageEnums is <Enum, Short> so we retrieve the first
```

Comment:
I really feel like this test knows too much about the implementation. I should ideally be able to go in and change out how we map things and the test cases should still pass if I've done it correctly. If the test expects biMap, that won't be true.